odoo/tools/template_inheritance.py

```diff
@@ -133,44 +133,64 @@ def extract(spec):
+                            source.set('t-name', t_name)
+                        if comment is not None:
+                            text = source.text
+                            source.text = None
+                            comment.tail = text
+                            source.insert(0, comment)
+                    else:
+                        replaced_node_tag = None
+                        for child in spec:
+                            if child.get('position') == 'move':
+                                child = extract(child)
+                            if inherit_branding and not replaced_node_tag and child.tag is not etree.Comment:
+                                # To make a correct branding, we need to
+                                # - know exactly which node has been replaced
+                                # - store it before anything else has altered the Tree
+                                # Do it exactly here :D
+                                child.set('meta-oe-xpath-replacing', node.tag)
+                                # We just store the replaced node tag on the first
+                                # child of the xpath replacing it
+                                replaced_node_tag = node.tag
+                            node.addprevious(child)
+                        node.getparent().remove(node)
+                elif mode == "inner":
+                    ''' Replace the entire content of an element '''
+                    node_children = list(node) # Target's children
+                    spec_children = list(spec) # Source's children
+                    for child in node_children:
+                        node.remove(child)
+                    node.text = None
+
+                    # Add xpath content
+                    if spec.text is not None:
+                        node.text = spec.text
+                    for child in spec_children:
+                        node.append(child)
                 else:
-                    replaced_node_tag = None
-                    for child in spec:
-                        if child.get('position') == 'move':
-                            child = extract(child)
-                        if inherit_branding and not replaced_node_tag and child.tag is not etree.Comment:
-                            # To make a correct branding, we need to
-                            # - know exactly which node has been replaced
-                            # - store it before anything else has altered the Tree
-                            # Do it exactly here :D
-                            child.set('meta-oe-xpath-replacing', node.tag)
-                            # We just store the replaced node tag on the first
-                            # child of the xpath replacing it
-                            replaced_node_tag = node.tag
-                        node.addprevious(child)
-                    node.getparent().remove(node)
+                    raise ValueError(
```

Comment:
@Esperax 

s alreayd discuss, It is the job of nrg to valid this kind of invalid view definition.
Thx to update it.

Reply:
My bad, we keep it like that ;)  without rng, and validation in python. It is the new trends.